dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -359,20 +389,47 @@ public String toJson(final Contentlet contentlet) throws DotMappingException {
                 mlowered.put(lowerCaseKey, lowerCaseValue);
 			}
 
-			final Optional<Field> binaryField = this.findFirstBinaryFieldIndexable(contentlet);
-			if (binaryField.isPresent()) {
+			if (Config.getBooleanProperty(WRITE_METADATA_ON_REINDEX, true)) {
 
-				this.generateBinaryMetadata(contentlet, sw, mlowered, binaryField.get());
+				if (contentlet.isLive() || contentlet.isWorking()) {
 
-				if (BaseContentType.FILEASSET.equals(contentlet.getContentType().baseType())) {
-					// see if we have content metadata
-					final File contentMeta = APILocator.getFileAssetAPI().getContentMetadataFile(contentlet.getInode());
-					if (contentMeta.exists() && contentMeta.length() > 0) {
-						final String contentData = APILocator.getFileAssetAPI().getContentMetadataAsString(contentMeta);
-						mlowered.put(FileAssetAPI.META_DATA_FIELD.toLowerCase() + StringPool.PERIOD + "content", contentData);
-					}
+					final ContentletMetadata metadata = fileMetadataAPI
```

Comment:
can all this logic be extracted to a private method?

Reply:
sure.. done!